lib/rubygems/text.rb

```diff
@@ -86,7 +86,7 @@ def levenshtein_distance(str1, str2)
 end
 
 # Use the latest and greatest DidYouMean::Levenshtein if it is available
-begin
+if RUBY_VERSION >= '2.7' && defined?(DidYouMean)
   require 'did_you_mean/levenshtein'
```

Comment:
My instincts are telling me that we should have this explicit require in here. 

1. `did_you_mean/levenshtein` is an implicit dependency of some of the did_you_mean spellcheckers. I don't like the idea of ruby retiring the use of levenshtein in the spell checker required by default and then we lose this implicit dependency. I like the idea of this failing quickly if in the future the environment is not what we expect. Avoid silent fallbacks and whatnot.